### PR TITLE
Makes CNI config more flexible

### PR DESCRIPTION
**Pull request by jordan-rash**

### CODE REVIEW

**Review by autodidaddict — APPROVED** [review]

File: internal/node/config.go [hunk 1]
```diff
@@ -22,6 +22,11 @@ func LoadNodeConfiguration(configFilepath string) (*models.NodeConfiguration, er
 		return nil, err
 	}
 
+	// TODO: should we validate the configuration here
```
Comment (kthomas) [inline internal/node/config.go:25]: I don't think so. We should be validating it in `NewNode`, and it returns an error if that validation fails.

See https://github.com/synadia-io/nex/blob/main/internal/node/node.go#L75

Comment (jordan-rash) [reply internal/node/config.go:25]: maybe both?  why wouldnt we validate it right after its created?

Comment (kthomas) [reply internal/node/config.go:25]: It wouldn't hurt anything if we did it inside of `LoadNodeConfiguration` sure -- but if so, do it at the very end before you return the config.

**Comment by kthomas** [comment]
This just needs to be rebased on main to fix the linting issue.

File: internal/node/config.go (hunk 1 above)
Comment (jordan-rash) [reply internal/node/config.go:25]: 👍🏼

**Review by autodidaddict — APPROVED** [review]

**Review by kthomas — APPROVED** [review]

**Pull request merged**